webnn/resources/test_data/slice.json

```diff
@@ -0,0 +1,772 @@
+{
+  "tests": [
+    {
+      "name": "slice float32 1D tensor default axes options",
+      "inputs": {
+        "input": {
+          "shape": [24],
+          "data": [
+            28.846251144212147,
+            97.9541470229301,
+            -68.15961736262523,
+            14.978987588557175,
+            90.23090084798065,
+            76.59095547712559,
+            -24.556316258204532,
+            79.58749302462488,
+            65.21375982234204,
+            57.43970862960305,
+            74.41775256222849,
+            -4.513182026141791,
+            0.5424534704526991,
+            80.44634130202977,
+            28.327650022124686,
+            74.0261938866893,
+            -74.54558964005362,
+            -27.30604081001799,
+            -70.4277433212161,
+            59.82631931755259,
+            -58.46095416849133,
+            79.80571087632629,
+            -9.857854207596304,
+            42.66519880465225
+          ],
+          "type": "float32"
+        }
+      },
+      "starts": [12],
+      "sizes": [12],
+      "expected": {
+        "name": "output",
+        "shape": [12],
+        "data": [
+          0.5424534678459167,
+          80.44634246826172,
+          28.32765007019043,
+          74.02619171142578,
+          -74.54559326171875,
+          -27.306041717529297,
+          -70.42774200439453,
+          59.82632064819336,
+          -58.46095275878906,
+          79.80570983886719,
+          -9.857853889465332,
+          42.665199279785156
+        ],
+        "type": "float32"
+      }
+    },
+    {
+      "name": "slice float32 2D tensor default axes options",
+      "inputs": {
+        "input": {
+          "shape": [4, 6],
+          "data": [
+            28.846251144212147,
+            97.9541470229301,
+            -68.15961736262523,
+            14.978987588557175,
+            90.23090084798065,
+            76.59095547712559,
+            -24.556316258204532,
+            79.58749302462488,
+            65.21375982234204,
+            57.43970862960305,
+            74.41775256222849,
+            -4.513182026141791,
+            0.5424534704526991,
+            80.44634130202977,
+            28.327650022124686,
+            74.0261938866893,
+            -74.54558964005362,
+            -27.30604081001799,
+            -70.4277433212161,
+            59.82631931755259,
+            -58.46095416849133,
+            79.80571087632629,
+            -9.857854207596304,
+            42.66519880465225
+          ],
+          "type": "float32"
+        }
+      },
+      "starts": [2, 2],
+      "sizes": [2, 4],
```

Comment:
It's interesting that WebNN chose `starts` and `sizes`, whereas most other frameworks I'm familiar with chose `starts` and `ends`, which makes it easier to specify relative endpoints (e.g. from element 0 to element N-1). Shrug. (no action expected)

Reply:
Yes. Some use `ends`/`stop` like ONNX ([Slice](https://github.com/onnx/onnx/blob/main/docs/Operators.md#Slice)) , OpenVINO ([Slice](https://docs.openvino.ai/latest/openvino_docs_ops_movement_Slice_8.html)), some use `sizes`/`size`, like DML ([SLICE](https://learn.microsoft.com/en-us/windows/win32/api/directml/ns-directml-dml_slice_operator_desc)),  TensorFlow ([slice](https://js.tensorflow.org/api/latest/#slice)).

Reply:
Plus:
- NumPy begin,end (https://numpy.org/doc/stable/user/basics.indexing.html?highlight=slice#slicing-and-striding
- PyTorch start,size (narrow)(https://pytorch.org/docs/stable/generated/torch.narrow.html#torch.narrow)